File: android/src/main/kotlin/com/screenmeet/sdk_live_flutter_plugin/handlers/ParticipantsStreamHandler.kt
package com.screenmeet.sdk_live_flutter_plugin.handlers

import com.screenmeet.sdk.Participant
import com.screenmeet.sdk_live_flutter_plugin.FlutterRTCVideoRenderer
import com.screenmeet.sdk_live_flutter_plugin.zipParticipants
import io.flutter.plugin.common.EventChannel

class ParticipantsStreamHandler: EventChannel.StreamHandler {

    private var sink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) { sink = events }
    override fun onCancel(arguments: Any?) { sink = null }

    fun sendParticipants(participants: List<Participant>, renderers: Map<String, FlutterRTCVideoRenderer>) {
        sink?.success(zipParticipants(participants, renderers))
    }
}

File: ios/Classes/StreamHandlers/ParticipantsStreamHandler.swift
//
//  ParticipantsStreamHandler.swift
//  sdk_live_flutter_plugin
//
//  Created by Ross on 01.11.2021.
//

import Foundation
import Flutter
import ScreenMeetSDK

class ParticipantsStreamHandler: NSObject, FlutterStreamHandler {
    private let pm = SwiftChannelParamsManager()

    var sink: FlutterEventSink?
       
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
       
    func sendParticipants(_ participants: [SMParticipant], _ renderers: [String: FlutterRTCVideoRenderer]) {
        guard let sink = sink else { return }
        sink(pm.participantsToDict(participants, renderers))
    }
       
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}
